sky/resources.py

```diff
@@ -131,6 +132,11 @@ def __init__(
           _docker_login_config: the docker configuration to use. This include
             the docker username, password, and registry server. If None, skip
             docker login.
+          _requires_fuse: whether the task requires FUSE mounting support. This
+            is used internally by certain cloud implementations to do additional
+            setup for FUSE mounting. This flag also safeguards against using
+            FUSE mounting on existing clusters that do not support it. If None,
+            defaults to False.
```

Comment:
If we default to False, this means all previously launched and serialized Resources have this field default to False. Then let's say the same YAML used to launch the previous cluster is used in `launch -c existing`, does that mean 

https://github.com/skypilot-org/skypilot/blob/11650578d666641175d46a53843b6f36a08d1a59/sky/backends/cloud_vm_ray_backend.py#L2564-L2571

will fail?

Reply:
ah good catch - showed up in backward compatibility tests. fixed by changing the deserialization to default to False when reading from old clusters.

Reply:
Ok I see why this confused me. This code here will not be reached for the case I had in mind (non-k8s) due to less_demanding_than() call above, as inside that call we only check for k8s. I'd say add a comment on L2564 here to say "Will not be reached for non-k8s". (If it were reached, we would print this resource mismatch error.)